File: backend/GraficaModerna.API/Controllers/CouponsController.cs
using GraficaModerna.Application.Interfaces;
using Microsoft.AspNetCore.Mvc;

namespace GraficaModerna.API.Controllers;

[Route("api/[controller]")]
[ApiController]
public class CouponsController(ICouponService service, IContentService contentService) : ControllerBase
{
    private readonly ICouponService _service = service;
    private readonly IContentService _contentService = contentService;

    [HttpGet("validate/{code}")]
    public async Task<IActionResult> Validate(string code)
    {
        var settings = await _contentService.GetSettingsAsync();
        if (settings.TryGetValue("purchase_enabled", out var enabled) && enabled == "false")
            return BadRequest("Uso de cupons indisponível temporariamente.");

        var coupon = await _service.GetValidCouponAsync(code);
        if (coupon == null) return NotFound("Cupom inválido ou expirado.");

        return Ok(new { coupon.Code, coupon.DiscountPercentage });
    }
}

File: grafica-moderna/app/Http/Controllers/CouponsController.php
<?php

namespace App\Http\Controllers;

use App\Services\CouponService;
use App\Services\ContentService;

class CouponsController extends Controller
{
    protected CouponService $service;
    protected ContentService $contentService;

    public function __construct(CouponService $service, ContentService $contentService)
    {
        $this->service = $service;
        $this->contentService = $contentService;
    }

    public function validateCode(string $code)
    {
        $settings = $this->contentService->getSettings();
        if (isset($settings['purchase_enabled']) && $settings['purchase_enabled'] === 'false') {
            return response()->json(
                ['message' => 'Uso de cupons indisponível temporariamente.'],
                400
            );
        }

        $coupon = $this->service->getValidCoupon($code);

        if (!$coupon) {
            return response()->json(
                ['message' => 'Cupom inválido ou expirado.'],
                404
            );
        }

        return response()->json([
            'code' => $coupon->code,
            'discountPercentage' => $coupon->discount_percentage,
        ]);
    }
}